packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -59,6 +60,85 @@ function getVariationNameFromClass( className, registeredStyles = [] ) {
 	return null;
 }
 
+/**
+ * This hook is used to generate new block style variation overrides
+ * based on an incoming theme config. If a matching style is found in the config,
+ * a new override is created and returned.
+ *
+ * @param {Object} config A global styles object, containing settings and styles.
+ * @return {Array} An array of new block variation overrides.
+ */
+export function useUpdateBlockStyleVariationOverridesWithConfig( config ) {
+	const overrides = useSelect(
+		( select ) => unlock( select( blockEditorStore ) ).getStyleOverrides(),
+		[]
+	);
+	const { getBlockStyles } = useSelect( blocksStore );
```

Comment:
There are not called within a callback, so they should be within the subscription callback?

Reply:
Oh sorry, I was referring to `getBlockName` below, not `getBlockStyles` 🤦🏻

 I'll move it up into the callback. Thanks!